### PR TITLE
#1 - Added WebSection for Search App in the Desktop Surface

### PR DESCRIPTION
**Pull request by habib535**
Related to #1 

Added WebSection in Desktop Surface at the top bar to include text box from Search App.

### CODE REVIEW

File: src/SampleWebsiteTheme/Api/MainHandlers.cs [hunk 1]
```diff
@@ -72,6 +72,7 @@ private static void CreateDesktopLayout()
 
             WebSection appTitleBlendingPoint = CreateBlendingPoint(desktopSurface, "AppTitle");
             WebSection mainMenuBlendingPoint = CreateBlendingPoint(desktopSurface, "MainMenu");
+            WebSection searchBlendingPoint = CreateBlendingPoint(desktopSurface, "Search");
```
Comment (warpech) [inline src/SampleWebsiteTheme/Api/MainHandlers.cs]: I believe that this should not be a new blending point. The existing blending points:

```
WebSection appTitleBlendingPoint = CreateBlendingPoint(desktopSurface, "AppTitle");
WebSection mainMenuBlendingPoint = CreateBlendingPoint(desktopSurface, "MainMenu");
WebSection logoBlendingPoint = CreateBlendingPoint(desktopSurface, "Logo");
WebSection topBarBlendingPoint = CreateBlendingPoint(desktopSurface, "TopBar");
```

should be renamed to more general names, respectively:

```
WebSection topBarLeftBlendingPoint = CreateBlendingPoint(desktopSurface, "Top Bar Left");
WebSection topBarLeftCornerBlendingPoint = CreateBlendingPoint(desktopSurface, "Top Bar Left Corner");
WebSection topBarCenterBlendingPoint = CreateBlendingPoint(desktopSurface, "Top Bar Center");
WebSection topBarRightBlendingPoint = CreateBlendingPoint(desktopSurface, "Top Bar Right");
```

Similarly, rename the `slot`s in HTML.

`topBarCenterBlendingPoint` should contain both the logo and the search pinning rules.

WDYT?

desktop-surface

**Comment by warpech** [comment]
Ping @habib535

**Comment by habib535** [comment]
I've added the Search and Logo inside a single Section. Now the problem is that they are appearing on top of each other, I want to re position them by applying some css styles but I am unable to find the right selector inside shadow dom. 

Here are the steps to reproduce

Set samplewebsitetheme to `develop-add-search` branch while the rest on develop branch

```
call samplewebsitetheme\run.bat
call signin\run.bat
call website\run.bat
call launchpad\run.bat
call images\run.bat
call search\run.bat
```
image

**Comment by warpech** [comment]
To fix the problem on the above screenshot, you need to move the styles for the toolbars out of Declarative Shadow DOM, unfortunately. Like @tomalec did in case of DefaultSurface here: https://github.com/StarcounterApps/Website/commit/e5958cd5600775c84bbda855a053949c158369a6#diff-93a1acc2001e27517a1f86816ab3af74

That was part of https://github.com/StarcounterApps/Website/pull/80.

@tomalec can you give some explanation why exactly this needs to be in DOM, not in Shadow DOM?

**Comment by warpech** [comment]
@habib535, what's the status of this?

**Comment by tomalec** [comment]
> @tomalec can you give some explanation why exactly this needs to be in DOM, not in Shadow DOM?

Could you specify "this"? in the commit you provided, I cannot see styles moved from Shadow to Light DOM.

I remember we needed some styles in Light DOM, to be able to select `::content > starcounter-composition > juicy-composition` to change style of nested partial view blending/insertion point.

But this is the thing I'm currently fixing, by removing `juicy-composition` so you should be able to use `::slotted(starcounter-composition)` and that should be sufficient to select the host of nested partial.

**Comment by habib535** [comment]
Hi @warpech,  moving styles out of the declarative shadow dom fixed the issue. Can you please review the updated changes? Thanks.

**Comment by habib535** [comment]
Before moving out the styles, I tried `::slotted` but that was only allowing to select the first level host. I had to apply styles to the third level.

**Comment by warpech** [comment]
> Could you specify "this"? in the commit you provided, I cannot see styles moved from Shadow to Light DOM.

Sorry, I have linked the wrong commit. I mean this commit https://github.com/StarcounterApps/Website/commit/4f4b5ddc3925762dd5de556ca3da36ddde2bd3b7, in which you have moved styles in file `src/Website/wwwroot/Website/templates/DefaultTemplate.html` from Shadow DOM to light DOM.

File: src/SampleWebsiteTheme/wwwroot/SampleWebsiteTheme/html/DesktopSurface.html [hunk 2]
```diff
@@ -15,13 +15,18 @@
         .desktop-theme-button__icon {
             cursor: pointer;
         }
+
+        .desktop-theme__top-center > juicy-composition > starcounter-include,
+        .desktop-theme__top-center > juicy-composition > starcounter-include > juicy-composition {
+            display: flex;
+        }
```
Comment (warpech) [inline src/SampleWebsiteTheme/wwwroot/SampleWebsiteTheme/html/DesktopSurface.html]: The selector looks wrong. And it misses some alignment and justification rules. Compare with:

https://github.com/StarcounterApps/Website/blob/develop/src/WebsiteEditor/wwwroot/WebsiteEditor/surfaces/DefaultSurface.html#L6-L11

**Comment by tomalec** [comment]
Maybe it's better to wait for  `juicy-composition`-less `starcounter-include` with this?

**Comment by warpech** [comment]
> Maybe it's better to wait for  `juicy-composition`-less `starcounter-include` with this?

No, this has to utilise whatever we have working now, because the project that requires it starts soon. When we have `juicy-composition`-less `starcounter-include`, we will update all of the apps.

**Pull request merged**